src/App.tsx

```diff
@@ -1,32 +1,46 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
+import { useEffect, useState } from 'react'
 import './App.css'
+import { CartPage } from './domain/cart/CartPage'
+import { BooksPage } from './domain/books/BooksPage'
+import Navbar from './components/navbar/navbar'
+import { getBooks, useAppDispatch, useAppSelector } from './domain/books/booksSlice'
+import { Book } from './domain/books/types'
+import { addToCart,removeFromCart} from './domain/cart/cartSlice'
+import { CartItem, CartState } from './domain/cart/types'
 
 function App() {
-  const [count, setCount] = useState(0)
+  const [route, setRoute] = useState<number>(0)
+  const dispatch = useAppDispatch()
+  const { books,loading } = useAppSelector(state => state.books)
+  const { cartItems } = useAppSelector(state => state.cart)
+
+useEffect(() => {
+  dispatch(getBooks())
+}, [])
+
+  const changeRoute = (x: number): void => {
+    setRoute(x)
+  }
+  const handleAddToCart = (item:Book): void=>{
+    console.log(item)
+    dispatch(addToCart(item))
+  }
+  const handleRemoveCartItem = (item:CartItem):void=>{
+    dispatch(removeFromCart(item))
+  }
+  const getCartTotalCurrency = ({cartItems}: CartState):number=>{
+    let total :number= 0
+    cartItems.forEach((item:CartItem)=> total +=item?.price)
+    return total;
+  }
```

Comment:
No need to write the fn here and pass it as a prop. Let Cart page handle itself.